app/logical/sources/strategies/tumblr.rb

```diff
@@ -161,14 +168,22 @@ def dtext_artist_commentary_desc
     # http://media.tumblr.com/tumblr_m24kbxqKAX1rszquso1_1280.jpg
     # => https://media.tumblr.com/tumblr_m24kbxqKAX1rszquso1_1280.jpg
     def find_largest(url, sizes: SIZES)
-      return url unless url =~ OLD_IMAGE
+      if url =~ OLD_IMAGE
+        candidates = sizes.map do |size|
+          "https://media.tumblr.com/#{$~[:dir]}#{$~[:filename]}_#{size}.#{$~[:ext]}"
+        end
 
-      candidates = sizes.map do |size|
-        "https://media.tumblr.com/#{$~[:dir]}#{$~[:filename]}_#{size}.#{$~[:ext]}"
-      end
+        candidates.find do |candidate|
+          http_exists?(candidate)
+        end
+      elsif url =~ %r{/s\d+x\d+/(\w+\.\w+)$}i
+        max_size = Integer.sqrt(Danbooru.config.max_image_resolution)
+        url = url.gsub(%r{/s\d+x\d+/\w+\.\w+$}i, "/s#{max_size}x#{max_size}/#{$1}")
 
-      candidates.find do |candidate|
-        http_exists?(candidate)
+        resp = Danbooru::Http.cache(1.minute).get(url, headers: NEW_HEADERS).parse
```

Comment:
The base strategy defines a `http` object that should be used by source strategies instead of instantiating a new `Danbooru::Http` client themselves. This is so that default http settings are shared by all source strategies.

Also the user agent is already set inside `Danbooru::Http#initialize`.